launch_xml/launch_xml/entity.py

```diff
@@ -55,6 +56,28 @@ def children(self) -> List['Entity']:
         """Get the Entity's children."""
         return [Entity(item) for item in self.__xml_element]
 
+    def assert_no_children(self):
+        if len(self.__xml_element):
+            raise ValueError(
+                f'Entity `{self.type_name}`` should not have any nested entity,'
+                f'but the following were found: {tuple(item.tag for item in self.__xml_element)}'
+            )
+
+    def assert_subentity_types(self, types: Iterable[str]):
+        for item in self.__xml_element:
+            if item.tag not in types:
+                raise ValueError(
+                    f'Found subentity of type `{item.tag}` in a `{self.__xml_element.tag}`, '
+                    f'which expects only the following subentities: {types}'
+                )
```

Comment:
@ivanpauno hmm, do we need this distinction between children and subentities? IIRC all of them are (sub)entities, whether we require them to be of an specific type or not.

Reply:
They aren't exactly the same in launch yaml AFAIR:

```yaml
launch:
  group:
    - executable:
      env:
        - ...
    - executable: ...
```